### ISSUE TITLE
Using Augmentor + TF Keras on greyscale images

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#288]
keep running into image shape problems

Say my image is greyscale...1 channel, batch size 10, 28x28
```
image_0 = tf.constant(np.random.random((10,28,28)))
image_1 = tf.constant(np.random.random((10,28,28,1)))

layer = tf.keras.layers.Conv2D(filters=2, kernel_size=(5,5))

output = layer(image_0) # <- will fail
output = layer(image_1) # <- will work!
```

So then our input image should be of shape (batchsize, row, height, 1)

BUUUT, if we use Augmentor Pipeline and then use keras_preprocess_func() to pass into tensorflow's image gen stuff, the first line that get's called is
```image = Image.fromarray(np.uint8(255 * image))``` where *Image* is ``from PIL import Image```.
That will fail when there is only 1 channel and that shape gives that channel...
```
# NOTE only handles one image at a time
image_0 = np.random.random((28,28))
image_1 = np.random.random((28,28,1))

from PIL import Image

image = Image.fromarray(np.uint8(255 * image_0)) # <- will work!
image = Image.fromarray(np.uint8(255 * image_1)) # <- will fail

**Comment by roddtalebi** [comment]
And that doesn't even begin to cover the issue with how my data may start as np.uint8 but then by the time it goes to Augmentor.Pipeline.keras_preprocess_func it becomes a float ...wtf 

Okay, i figured that part out...you have to pass dtype=np.uint8 when you make the imagedatagen
https://github.com/keras-team/keras/blob/v2.8.0/keras/preprocessing/image.py#L807

BUT that still brings up the question as to why the Augmentor Pipeline keras_preprocess_func() expects that the image was already normalized

**Comment by roddtalebi** [comment]
```
image = (np.random.random((28,28,3))*255).astype(np.uint8)
new_image = np.array(255*image).astype(np.uint8)
np.all(new_image==image) # <- returns False
```

that only reiterates that they for some reason expect Augmentor Pipeline keras_preprocess_func to be an image with values between [0,1] but can't say why they assume that

**Comment by roddtalebi** [comment]
Long story short:

- if using greyscale or other single channel images, don't use Augmentor Pipeline
- if using Augmentor Pipeline, make sure the data is normalized
